jerry-libc/jerry-libc.c

```diff
@@ -263,7 +240,8 @@ strlen (const char *s)
 int
 rand (void)
 {
-  uint32_t intermediate = libc_random_gen_state[0] ^ (libc_random_gen_state[0] << 11);
+  uint32_t intermediate = libc_random_gen_state[0];
+  intermediate ^= intermediate << 11;
```

Comment:
This is harder to me to read. And I think the compiler can optimize this as well.